usr/share/rear/prep/default/400_save_directories.sh

```diff
@@ -0,0 +1,101 @@
+#
+# usr/share/rear/prep/default/400_save_directories.sh
+#
+# Purpose of this script is to save permissions, owner, group
+# or symbolic link name and target of basic directories in a file.
+# That file will be read during recovery time to recreate those directories.
+# The script restore/default/900_create_missing_directories.sh will recreate
+# those directories if they were not already recreated (e.g. via backup restore)
+# and all other code or scripts that also does this could/should be deleted.
+
+local directories_permissions_owner_group_file="$VAR_DIR/recovery/directories_permissions_owner_group"
+cat /dev/null >"$directories_permissions_owner_group_file"
+
+# First save directories that are used as mountpoints:
+# BUILD_DIR can be used in 'grep -v "...|$BUILD_DIR|..."' because it is never empty (see usr/sbin/rear)
+# because with any empty part 'grep  -v "...||..."' would output nothing at all:
+local mountpoints="$( mount | grep -vE "cgroup|fuse|nfsd|/sys/|$BUILD_DIR|REAR-000" | awk '{print $3}' )"
```

Comment:
maybe add cifs ?

Reply:
I would love to understand the reasons behind
why each particular stuff is excluded.

E.g. why mountpoint directories for 'fuse' and 'nfsd'
are not kept?
In particular I wonder why any entries that contains
'fuse' or 'nfsd' anywhere on the line are excluded
e.g. also entries like
<pre>
/dev/sdb7 on /stuffusage type ext3
/dev/sdb8 on /nfsdata type ext4
</pre>

I would expect to have mountpoint directories
for 'fuse' and 'nfsd' back after "rear recover"
so that one can mount 'fuse' and 'nfsd' stuff there.
But there must be some obscure reason why those
mountpoint directories must not be re-created.
I am neither a fuse nor a nfsd expert so that I
do not "just know" the resons and I even like
to play dumb here and "just keep it as I found it"
without wasting too much time with too much wondering
about each and every piece of obscure code that
seems to "just work" but I only don't know why.

Reply:
Those directories will be back after recovery as they are described in the disklayout.conf file, no?
The main purpose was to reduce the list with moint points used by ReaR itself, and some built-in system mount point which are created always by the kernel layer.

Reply:
@jsmeix 
You've made a good point with:

> In particular I wonder why any entries that contains
> 'fuse' or 'nfsd' anywhere on the line are excluded
> e.g. also entries like

A particular solution would be to put prefix "type" in front of  `grep` ...

However this starts look like start of bloatware code ...
To exclude all "noise" from mounts on my Debian Stretch, I had to use following:

```
mount | grep -vE "type (cgroup|tmpfs|vfat|hugetlbfs|binfmt_misc|fusectl|fuse.gvfsd-fuse|autofs|sysfs|proc|devtmpfs|devpts|securityfs|pstore|efivarfs|mqueue|debugfs) "
```

So this part is IMHO great candidate for further issues ...

V.

Reply:
For now I mark it with a FIXME comment.

Reply:
@gozora I wouldn't exclude `vfat`, might be needed for UEFI and others.

Suggestion:
```
findmnt -runit 'autofs,binfmt_misc,cgroup,cgroup2,configfs,debugfs,devpts,devtmpfs,efivarfs,fusectl,fuse.gvfs-fuse-daemon,hugetlbfs,mqueue,nfsd,proc,pstore,securityfs,sysfs,tmpfs' -o TARGET
``

Reply:
@jsmeix @gozora @ProBackup-nl I agree **vfat** should not be excluded.